File: sources/js/list/SqList_test.js
const SqList = require('./SqList')

const l = new SqList()
l.display()

l.insert(11, 1)
l.insert(12, 2)
l.insert(13, 3)
l.insert(14, 4)
l.insert(15, 5)
l.insert(17, 7)
l.insert(15, 5)
l.display()

console.log('删除元素:', l.delete(3))
l.display()

l.update(1, 100)
l.display()

console.log('查询元素:', l.search(2))

console.log('查询元素:', l.locate(100))

File: sources/c/list/SqList_test.c
#include <stdio.h>
#include "SqList.c"

int main(){
    // 初始化一个顺序表
    SqList L = newSqList();
    display(&L);

    insert(&L, 11, 1);
    insert(&L, 12, 2);    
    insert(&L, 13, 3);
    insert(&L, 14, 4);
    insert(&L, 15, 5);
    insert(&L, 17, 7);
    insert(&L, 15, 5);
    display(&L);

    int e1;
    delete(&L, 3, &e1);
    printf("删除元素：%d\n", e1);
    display(&L);

    update(&L, 1, 100);
    display(&L);

    int e2;
    search(&L, &e2, 2);
    printf("查询元素：%d\n", e2);

    printf("查询元素：%d\n", locate(&L, 100));

    return 0;
}